### PR TITLE
feat integrate planarfreehand tool to app. Including SR load

### PR DESCRIPTION
**Pull request by ladeirarodolfo**
### PR Checklist
- It integrates planar freehand tool to app
- It also includes SR loading
- Free contribution (but it will be needed soon for at least one client)

- [ ] Brief description of changes
- [ ] Links to any relevant issues
- [ ] Required status checks are passing
- [ ] User cases if changes impact the user's experience
- [ ] `@mention` a maintainer to request a review

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: platform/docs/versioned_docs/version-3.0/platform/services/data/ToolbarService.md [hunk 1]
```diff
@@ -210,6 +210,16 @@ to create `MeasurementTools` nested button
...
+      // Planar Freehand tool
+      {
+        id: 'PlanarFreehandROI',
+        icon: 'tool-freehand',
+        label: 'Planar Freehand',
+        type: 'tool',
+        commandOptions: {
+          toolName: 'PlanarFreehandROI',
+        }
+      },
```
Comment (sedghi) [inline platform/docs/versioned_docs/version-3.0/platform/services/data/ToolbarService.md:222]: why are you chaning versioned docs? these are archived, you need to change the root docs

Comment (ladeirarodolfo) [reply platform/docs/versioned_docs/version-3.0/platform/services/data/ToolbarService.md:222]: :D ok thanks

File: modes/tmtv/src/initToolGroups.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ function initToolGroups(toolNames, Enums, ToolGroupService, commandsManager) {
       { toolName: toolNames.DragProbe },
       { toolName: toolNames.EllipticalROI },
       { toolName: toolNames.RectangleROI },
+      { toolName: toolNames.PlanarFreehandROI },
```
Comment (sedghi) [inline modes/tmtv/src/initToolGroups.js:33]: tmtv mode does not need freehand tool since there is no measurement tracking/export etc

**Review by sedghi — CHANGES_REQUESTED** [review]
I put minor comments in the code but major ones are 

1. Where did we get the svg icon? I need to ask Dan to design a freehand tool icon
image

2. After exporting the SR it shows some weird statistics see below

image

3. reimporting back the SR (refresh and then drag and drop gain) gives error 
 
image

File: modes/tmtv/src/initToolGroups.js (hunk 2 above)
Comment (ladeirarodolfo) [reply modes/tmtv/src/initToolGroups.js:33]: ok

**Comment by ladeirarodolfo** [comment]
> I put minor comments in the code but major ones are
> 
> 1. Where did we get the svg icon? I need to ask Dan to design a freehand tool icon
>    image
> 2. After exporting the SR it shows some weird statistics see below
> 
> image
> 
> 3. reimporting back the SR (refresh and then drag and drop gain) gives error
> 
> image



> I put minor comments in the code but major ones are
> 
> 1. Where did we get the svg icon? I need to ask Dan to design a freehand tool icon
>    image
> 2. After exporting the SR it shows some weird statistics see below
> 
> image
> 
> 3. reimporting back the SR (refresh and then drag and drop gain) gives error
> 
> image

1. I have done it manually

2. Ok I will take a look

**Comment by ladeirarodolfo** [comment]
@sedghi  I have covered the items as following:

1. Icon svg, using existing on PR
2. After exporting SR issue: fixed it
3. Reimport SR issue. This a PACS configuration, probably you are using dcm4chee-arc with a config that tells for "FL" value to replace by bulkdata uri.
You can double check that if you draw a small Freehand. I validated here on my env using different PACS system and also different dcm4chee configuration. So I`m not sure if this effort should be part of this task


While doing some tests I realized a "bug" on annotate tool which I logged here
https://github.com/OHIF/Viewers/issues/2937

**Comment by sedghi** [comment]
> 3\. This a PACS configuration, probably you are using dcm4chee-arc with a config that tells for "FL" value to replace by bulkdata uri.

I used the QA server which is Google Healthcare API, why it is PACS dependant?

**Comment by ladeirarodolfo** [comment]
Let me formulate better.
Most Archiving designs has the possibility of limiting data to reduce memory footprint on storage.
In that, there is the strategy is to limit the value of representation limit.
For dcm4chee is here
https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/bulkData.html#dcmbulkdatavrlengththreshold

For google healthcare, specifically, I couldnt access the google console of OHIF configuration (I believe I lost the rights), to double check it. Also my user can not save SRs. I believe google heatlhcare might have similar limitation as dcm4chee and other archiving systems. I will more than happy to troubleshoot it if you guarantee me access to save SR or access to google console.

**Comment by sedghi** [comment]
what is the PACS you are using that you see it works?

**Comment by sedghi** [comment]
So apparently the `graphicData` is returned undefined by the servers (most likely for the reason that you say), however, It should send a bulkDataURL so that we can request it similar to how we do here for [palleteColors](https://github.com/OHIF/Viewers/blob/cd21a80e1cfa16247f5ff7bceaee0a1fe34012e7/platform/core/src/utils/metadataProvider/fetchPaletteColorLookupTableData.js#L67). This seems to be a bug in OHIF and we should fix it first

**Comment by wayfarer3130** [comment]
I'm closing this in favour of a newer PR that includes the planar freehand tool as well as several others.

**Pull request closed without merging**